stm/stm.mli

```diff
@@ -169,6 +169,8 @@ val finish_tasks : string ->
   Library.seg_univ -> Library.seg_discharge -> Library.seg_proofs ->
     tasks -> Library.seg_univ * Library.seg_proofs
 
+val get_success_state : doc:doc -> Stateid.t -> Vernacstate.t option
```

Comment:
Adding a new API function and no documentation. Please document this properly.

Reply:
Right, I added some documentation now.

Reply:
Ah OK, you put the comments in the `.ml` file so I wasn't seeing it. I think the doc reads well, tho I am not sure why success states are more special than failed ones.

From the point of view of the client, I would expect that doing `(Check sid)` would eventually make `sid` to always have a state, either failed or successful. This extra logic complicates things for now reason IMHO.

Also, the magic going on about "if the current state is sid, do this" seems pretty weird to me, but that I guess you have a need for it. But I would expect states to be installed explicitly.